File: backend/app/models/partner_profile.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict


class ImportantDates(BaseModel):
    birthday: Optional[str] = None
    anniversary: Optional[str] = None
    other: Optional[Dict[str, str]] = None
    # example: { "first_date": "2023-04-10" }


class RelationshipContext(BaseModel):
    user_name: Optional[str] = None
    partner_name: Optional[str] = None
    relationship_stage: Optional[str] = None
    relationship_start_date: Optional[str] = None


class GrowthIntent(BaseModel):
    user_goals: Optional[List[str]] = None
    partner_expressed_needs: Optional[List[str]] = None
    relationship_vision: Optional[str] = None


class PartnerProfileV1(BaseModel):
    context: RelationshipContext = Field(...)

    pronouns: Optional[str] = None
    important_dates: Optional[ImportantDates] = None

    preferences: Optional[List[str]] = None
    love_languages: Optional[List[str]] = None
    boundaries: Optional[List[str]] = None

    growth_intent: Optional[GrowthIntent] = None

    notes: Optional[str] = None

File: types/partner-profile.ts
export interface RelationshipContext {
  user_name?: string;
  partner_name?: string;
  relationship_stage?: string;
  relationship_start_date?: string;
}

export interface ImportantDates {
  birthday?: string;
  anniversary?: string;
  other?: Record<string, string>;
}

export interface GrowthIntent {
  user_goals?: string[];
  partner_expressed_needs?: string[];
  relationship_vision?: string;
}

export interface PartnerProfile {
  context: RelationshipContext;
  pronouns?: string;
  important_dates?: ImportantDates;
  preferences?: string[];
  love_languages?: string[];
  boundaries?: string[];
  growth_intent?: GrowthIntent;
  notes?: string;
}
